modules/Sources/Features/NHSendFlow/NHScan/NHScanStore.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  NHScanStore.swift
+//  
+//
+//  Created by Matthew Watt on 7/22/23.
+//
+
+import CaptureDevice
+import ComposableArchitecture
+import URIParser
+import Utils
+import ZcashLightClientKit
+
+public typealias NHScanStore = Store<NHScanReducer.State, NHScanReducer.Action>
+public typealias NHScanViewStore = ViewStore<NHScanReducer.State, NHScanReducer.Action>
+
+public struct NHScanReducer: ReducerProtocol {
+    public struct State: Equatable {
+        public enum ScanStatus: Equatable {
+            case failed
+            case value(RedactableString)
+            case unknown
+        }
+        
+        public var scanStatus: ScanStatus = .unknown
+        public var scannedValue: String? {
+            guard case let .value(scannedValue) = scanStatus else {
+                return nil
+            }
+            
+            return scannedValue.data
+        }
+        
+        public init() {}
+    }
+    
+    public enum Action: Equatable {
+        case backButtonTapped
+        case onAppear
+        case onDisappear
+        case found(RedactableString)
+        case scanFailed
+        case scan(RedactableString)
+    }
+    
+    private enum CancelId { case timer }
+    
+    let networkType: NetworkType
+    @Dependency(\.captureDevice) var captureDevice
+    @Dependency(\.dismiss) var dismiss
+    @Dependency(\.mainQueue) var mainQueue
+    @Dependency(\.uriParser) var uriParser
+    
+    public var body: some ReducerProtocol<State, Action> {
+        Reduce { state, action in
+            switch action {
+            case .backButtonTapped:
+                return .run { _ in await self.dismiss() }
+            case .onAppear:
+                // reset the values
+                state.scanStatus = .unknown
```

Comment:
[non-blocing] I think this shouldn't clear up the values when the app is backgrounded and user returns to it relying on the values. but it's worth testing it.